portfolio/src/main/webapp/script.js

```diff
@@ -65,3 +65,186 @@ function addRandomFunFact() {
+    if (currentPage.firstElementChild.innerText == numPages - 2) { 
+      return; 
+    }
+
+    currentPage.nextSibling.classList.add('active');
+  } else {
+    event.currentTarget.classList.add('active');
+  }
+
+  currentPage.classList.remove('active');
+
+  // Fetch comments on that page without recreating pagination.
+  fetchComments(false);
+}
+
+/**
+ * Fetches the comments from the data server and displays the comments on the currently active page.
+ * @param {boolean=} createNewPagination Whether a new pagination bar should be created.
+ */
+function fetchComments(createNewPagination = false) {
+  fetch('/data').then(response => response.json()).then(comments => {
+    const commentsListElement = document.getElementById('comments-list');
+    commentsListElement.innerHTML = '';
+
+    const commentsPerPageElement = document.getElementById('num-comments-per-page');
+    const commentsPerPage = commentsPerPageElement.options[commentsPerPageElement.selectedIndex].value;
+
+    const currentPage = document.querySelector('.page-item.active');
+    var currentPageNumber = currentPage.firstElementChild.innerText;
+    if (createNewPagination) {
+      currentPageNumber = createPaginationBar(Math.ceil(comments.length / commentsPerPage), currentPageNumber);
+    }
+
+    const startIndex = (currentPageNumber - 1) * commentsPerPage;
+    const endIndex = Math.min(Number(commentsPerPage) + Number(startIndex), comments.length);
+    for (var i = startIndex; i < endIndex; i++) {
+      commentsListElement.appendChild(createCommentElement(comments[i]));
+    }
+  });
+}
+
+/**
+ * Creates an <li> element containing a comment, including the username, content, time, and delete button.
+ * @param {!Comment} comment The Comment object from which a <li> element is created.
+ * @return {!Element<li>} The comment element created.
+ */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const commentContainer = document.createElement('div');
+
+  const nameElement = document.createElement('strong');
+  nameElement.className = 'text-success';
+  nameElement.innerText = '@' + comment.name + ' - ' + comment.mood;
+  const timeElement = document.createElement('span');
+  timeElement.className = 'pull-right text-muted';  // Add Bootstrap classes to style the timeElement.
+  timeElement.innerText = convertToDateTime(comment.timestamp);
+  const contentElement = document.createElement('p');
+  contentElement.innerText = comment.content;
+  
+  commentContainer.appendChild(timeElement);
+  commentContainer.appendChild(nameElement);
+  commentContainer.appendChild(contentElement);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.innerText = 'Delete';
+  deleteButton.addEventListener('click', () => {
+    deleteComment(comment);
+    commentElement.remove();  // Remove the comment from the DOM.
```

Comment:
Does the use of "commentElement" here associate the element with this current comment correctly?
To be sure, you might have to do something like `deleteButton.parentNode` to get the correct direct parent of the button and then delete that element.

Reply:
Yes, "commentElement" does work correctly. But you've got a good point. I've changed it to `deleteButton.parentNode`, but still you can ask whether `deleteButton` is the correct button lol. I guess it's one of the differences between web app and android.

Reply:
Thanks for making the change.

Haha that's true!